pennylane/ops/qubit/matrix_ops.py

```diff
@@ -296,9 +319,92 @@ def compute_decomposition(U: TensorLike, wires: WiresLike):
                 raise DecompositionUndefinedError(
                     "The decomposition of a two-qubit sparse QubitUnitary is undefined."
                 )
+
             return qml.ops.two_qubit_decomposition(U, Wires(wires))
 
-        return super(QubitUnitary, QubitUnitary).compute_decomposition(U, wires=wires)
+        ops = []
```

Comment:
Can you move this logic to a separate function in `pennylane/ops/op_math/decompositions/unitary_decompositions.py`?